evcache-core/src/main/java/com/netflix/evcache/EVCacheImpl.java

```diff
@@ -1518,80 +1829,199 @@ public T get(long timeout, TimeUnit unit) throws InterruptedException, Execution
+        if (hasHashedKey) {
+            return CompletableFuture
+                    .supplyAsync(SupplierUtils.wrap(() -> client
+                            .getBulk(keyMap.keySet(),
+                                    evcacheValueTranscoder,
+                                    throwException,
+                                    hasZF)),
+                            executorService)
+                    .thenApply(data -> buildHashedKeyValueResult(data, tc, client, keyMap))
+                    .exceptionally(t -> handleBulkException(t, evcacheKeys, hasZF, throwException));
+        } else {
+            final Transcoder<T> tcCopy;
+            if (tc == null && _transcoder != null) {
+                tcCopy = (Transcoder<T>) _transcoder;
+            } else {
+                tcCopy = tc;
+            }
+            return CompletableFuture
+                    .supplyAsync(SupplierUtils.wrap(() -> client.getBulk(keyMap.keySet(),
+                            tcCopy ,
+                            throwException,
+                            hasZF)))
+                    .thenApply(data -> buildNonHashedKeyValueResult(data, keyMap))
+                    .exceptionally(t -> handleBulkException(t, evcacheKeys, hasZF, throwException));
+        }
+    }
+
+    private <T> Map<EVCacheKey, T> handleBulkException(Throwable t, Collection<EVCacheKey> evCacheKeys, boolean hasZF, boolean throwException) {
+        if (log.isDebugEnabled() && shouldLog())
+            log.debug("Exception while getBulk data for APP " + _appName + ", key : " + evCacheKeys, t);
+        if (!throwException || hasZF) return null;
+        throw Sneaky.sneakyThrow(t);
+    }
+
+    private KeyMapDto buildKeyMap(EVCacheClient client, Collection<EVCacheKey> evcacheKeys) {
+        boolean hasHashedKey = false;
+        final Map<String, EVCacheKey> keyMap = new HashMap<String, EVCacheKey>(evcacheKeys.size() * 2);
+        for (EVCacheKey evcKey : evcacheKeys) {
+            String key = evcKey.getCanonicalKey(client.isDuetClient());
+            String hashKey = evcKey.getHashKey(client.isDuetClient(), client.getHashingAlgorithm(), client.shouldEncodeHashKey(), client.getMaxDigestBytes(), client.getMaxHashLength(), client.getBaseEncoder());
+            if (hashKey != null) {
+                if (log.isDebugEnabled() && shouldLog())
+                    log.debug("APP " + _appName + ", key [" + key + "], has been hashed [" + hashKey + "]");
+                key = hashKey;
+                hasHashedKey = true;
+            }
+            keyMap.put(key, evcKey);
+        }
+        return new KeyMapDto(keyMap, hasHashedKey);
+    }
+
+    private <T>  Map<EVCacheKey, T>  buildNonHashedKeyValueResult(Map<String, T> objMap,
+                                                                  Map<String, EVCacheKey> keyMap) {
+        final Map<EVCacheKey, T> retMap = new HashMap<EVCacheKey, T>((int) (objMap.size() / 0.75) + 1);
```

Comment:
curious on 0.75. I see this is from existing code, but maybe good to add a comment about why 0.75